train.py

```diff
@@ -188,8 +189,10 @@ def train(params, args, local_rank, world_rank, world_size):
             logging.info('  Avg train loss=%f'%np.mean(tr_loss))
             args.tboard_writer.add_scalar('Loss/train', np.mean(tr_loss), iters)
             args.tboard_writer.add_scalar('Learning Rate', optimizer.param_groups[0]['lr'], iters)
-            args.tboard_writer.add_scalar('Avg iters per sec', step_count/(end-start), iters)
-            fig = generate_images([inp, tar, gen])
+            args.tboard_writer.add_scalar('Avg iters per sec', step_count/float(end-start), iters)
+            fig = generate_images([inp.to(torch.float32),
```

Comment:
converted in generate_images function

Reply:
that threw an error in BF16 AMP mode because numpy cannot deal with bf16. I fixed it by moving the data mapping out of the autocrats region.

Reply:
hmm but it's currently converted to float before it's converted to a numpy array. Leaving @pzharrington to resolve this since he has bf16 runs without errors.

Reply:
hmm but its converted to float before it becomes a numpy array currently. Leaving @pzharrington to resolve this since he had bf16 runs go through.

Reply:
I had multiple runs w/ bf16 go through without any errors from plotting  -- I recall Josh fixed this awhile ago in the `generate_images` routine, looks like that change did not get propagated to this branch